tools/pd-ctl/pdctl/command/operator.go

```diff
@@ -423,3 +430,34 @@ func parseUint64s(args []string) ([]uint64, error) {
 	}
 	return results, nil
 }
+
+func parseUint64sAndRoles(args []string) (uint64, []uint64, []string, error) {
+	regionID, err := strconv.ParseUint(args[0], 10, 64)
+	if err != nil {
+		return 0, nil, nil, errors.WithStack(err)
+	}
+
+	if _, err := strconv.ParseUint(args[2], 10, 64); err == nil {
+		// No peer roles provided in arguments
+		storeIDs, err := parseUint64s(args[1:])
+		return regionID, storeIDs, nil, err
+	}
+
+	storeIDs, peerRoles := make([]uint64, 0, len(args)), make([]string, 0, len(args))
```

Comment:
Why len(args)?

Reply:
I was trying to preallocate the slice. It seems that I misunderstood how `append` works in go, and I'm removing it. Sorry!